### PR TITLE
bugfix: Autohelp

### LINKED ISSUE #105: Case-sensitive tag names
Tag names are case-sensitive.
Example:
`st lernjava` does not give the same result as `st Lernjava`

This can be quite annoying at times, although the bot already tells you the right lowercase name.
My suggestion is to just make the comparison case-insensitive.

### LINKED ISSUE #113: Autohelp does not detect classes with superclasses as classes
```java
public class Stuff extends/implements Super {
}
``` 
above example is not recognized as regex wants `<modifier> class <Name> {`

### LINKED ISSUE #119: Raw tags can exceed char limit
@JohnnyJayJay WHY TEH HECK IS YOUR lernjava TAG SO LONG

Log: https://haste.devcord.xyz/yecemacoro

Credit: @Veteranfighter

### LINKED ISSUE #122: Disable reactions in Rat-chat without Rat role

### PR DESCRIPTION
**Pull request by DRSchlaubi**
- Ignore whitespaces in exception names
- Allow modifiers behind class name (extends, implements) (#113)
- Fix bug in class name parsing
- Ignore if caused by is prefix by something like `Caused by in exception
- Update to Kotlin 1.4
- Update to Gradle 6.6
- Make tag names case insensitive (#105)
- Remove un-permitted reactions in rat-chat (#122)
- Fix to long raw tags causing exception (#119)

### Github
Fixes #113 
Fixes #105 
Fixes #122 
Fixes #119

### CODE REVIEW

File: src/main/kotlin/com/github/devcordde/devcordbot/core/RatProtector.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * @param roleId the id of the rat role with bypass permissions
+ */
+class RatProtector(private val channelId: Long, private val roleId: Long) {
+
+    /**
+     * Removes "bad" reactions.
+     */
+    @EventSubscriber
+    fun onReactionAdd(event: MessageReactionAddEvent) {
+        if (event.user?.isBot == true || event.channel.idLong != channelId || event.member?.roles?.any { it.idLong == roleId } == true) return
```
Comment (jkoenig134) [inline src/main/kotlin/com/github/devcordde/devcordbot/core/RatProtector.kt]: A team member should be able to react - not every team member has the "rat" role.

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/github/devcordde/devcordbot/core/RatProtector.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+        @Suppress("ReplaceNotNullAssertionWithElvisReturn") // We have caching enabled so it cannot be null
+        event.reaction.removeReaction(event.user!!).queue()
+    }
+}
```
Comment (jkoenig134) [inline src/main/kotlin/com/github/devcordde/devcordbot/core/RatProtector.kt]: newline

Comment (DRSchlaubi) [reply src/main/kotlin/com/github/devcordde/devcordbot/core/RatProtector.kt]: ragequit

**Review by jkoenig134 — APPROVED** [review]
LGTM

**Review by daspositioned — APPROVED** [review]

**Pull request merged**